### ISSUE TITLE
Issue with "Express Tutorial Part 7: Deploying to production": cannot find module 'app/bin/www' when deployed to Heroku

### ISSUE THREAD

**Issue by geko83** [mdn/content#11517]
MDN URL: https://developer.mozilla.org/en-US/docs/Learn/Server-side/Express_Nodejs/deployment

#### What information was incorrect, unhelpful, or incomplete?
Which changes I need to apply in the application before to release to production.

#### Specific section or headline?
"Create and upload the website"
https://developer.mozilla.org/en-US/docs/Learn/Server-side/Express_Nodejs/deployment#create_and_upload_the_website

#### What did you expect to see?
Which types of error I could get once I have released to production and how I can solve them.

#### Did you test this? If so, how?
Yes, I tried to deploy the project, but I got an error when I try to open the page.
I got the error message by analysing the log "Cannot find module '/app/bin/www'".
![NodeJs_error](https://user-images.githubusercontent.com/96536962/147496164-407a50f7-fa23-4fd0-a335-811eee3cfc8f.PNG)

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/learn/server-side/express_nodejs/deployment`
* MDN URL: https://developer.mozilla.org/en-US/docs/Learn/Server-side/Express_Nodejs/deployment
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/learn/server-side/express_nodejs/deployment/index.md
* Last commit: https://github.com/mdn/content/commit/0f1f6f74804725f379465c09f698b88611292c90
* Document last modified: 2021-12-01T06:01:14.000Z

</details>

**Comment by hamishwillee** [comment]
There should be no errors if you have followed the instructions properly.

Looking at this I'd guess that your npm start is calling something wrong.  From the [skeleton website](https://developer.mozilla.org/en-US/docs/Learn/Server-side/Express_Nodejs/skeleton_website#enable_server_restart_on_file_changes) the `scripts` section of your package.json file should say something like:
```
  "scripts": {
      "start": "node ./bin/www",
      "devstart": "nodemon ./bin/www",
      "serverstart": "DEBUG=express-locallibrary-tutorial:* npm run devstart"
    },
```
This starts a file called `\bin\www`. I think the error is that it can't find the module named app. On my version of the file www this is near the top and says 
```
var app = require('../app');
```
So you should check if you have in the root of your project a file called app.js. 

I recommend you start by comparing your project on github to this one: https://github.com/mdn/express-locallibrary-tutorial

**Comment by geko83** [comment]
Yes, I followed the instructions correctly, but the issue was in the .gitignore file.
In this file I have the www file name, so when I commit and push the changes to Heroku this file and the bin folder were not committed.
I tried by forcing the addition of the bin folder to the repository and it worked successfully.
Is this correct? Or why Heroku didn't create the \bin\www file when the app is compiled?
Many thanks.
